waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -254,7 +254,9 @@
           restart-count (get-in pod [:status :containerStatuses 0 :restartCount] 0)
           instance-id (pod->instance-id pod restart-count)
           node-name (get-in pod [:spec :nodeName])
-          port0 (get-in pod [:spec :containers 0 :ports 0 :containerPort])
+          has-reverse-proxy? (contains? (get-in pod [:spec :containers]) 2)
```

Comment:
Instead of expecting the container to be at position `2`, let's check for a container with the presence of a specific annotation, e.g. `waiter/reverse-proxy`.